src/api/localStorage.js

```diff
@@ -0,0 +1,96 @@
+export class LocalStorage {
+  constructor() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  createUser(user) {
+    this.updateUsers();
+    user.id = this.generateId();
+    this.users.push(user);
+    this.setLocalStorageData();
+  }
+
+  setLocalStorageData() {
+    localStorage.setItem('UserAccounts', JSON.stringify(this.users));
+  }
+
+  updateUsers() {
+    this.users = JSON.parse(localStorage.getItem('UserAccounts')) || [];
+  }
+
+  getUser(id) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    return user;
+  }
+
+  getAllUsers() {
+    return this.users;
+  }
+
+  editUser(id, editName, editValue) {
+    try{
+      const editObj = {};
+      editObj[`${editName}`] = `${editValue}`;
+      const updatedUser = Object.assign(this.getUser(id), editObj);
+      this.deleteUser(id);
+      this.createUser(updatedUser);
+    } catch (err) {console.log('Error Editing User: ', err);}
+  }
+
+  replaceUser(id, update) {
+    const user = this.users.find(user => String(user.id) === String(id));
+    const replacedUser = Object.assign(user, update);
+    return replacedUser;
+  }
+
+  deleteUser(id) {
+    this.updateUsers();
+    this.users.forEach((user, index) => {
+      if (String(user.id) === String(id)) {
+        this.users.splice(index, 1);
+        this.setLocalStorageData();
+      }
+    });
+  }
+
+  deleteAllUsers() {
+    this.users = [];
+    localStorage.clear();
+  }
+
+  generateId() {
+    return Math.floor((Math.random() * 1000) + 1);
+  }
+
+  validateUser(user) {
+    const validationObject = {
+      _id: String,
+      name: String
+    }
+    const userObject = Object.getOwnPropertyNames(user);
+    const testObject = Object.getOwnPropertyNames(validationObject);
+    if (userObject.length != testObject.length) {
+      console.log('Number Of Props Do Not Match, User Invalid');
+      return false;
+    }
+
+    for (let i = 0; i < userObject.length; i++) {
```

Comment:
It's a little weird but this could be simplified
```suggestion
    return !Boolean(userObject.find(prop => userObject[prop] !== testObject[prop]));
```
- `.find()` loops through `userObject`. 
- if it finds one of the iterations passes `userObject[prop] !== testObject[prop]`, it'll return that value. if none of the iterations passes, it returns `undefined`
- `Boolean()` will convert _`someValueFromFind`_ into `true` and `undefined` to `false`, the opposite of what you need. So inverse it with `!` and return